src/main/java/core/mvc/tobe/handlermapping/custom/AnnotationHandlerMapping.java

```diff
@@ -0,0 +1,82 @@
+package core.mvc.tobe.handlermapping.custom;
+
+import com.google.common.collect.Maps;
+import core.annotation.web.Controller;
+import core.annotation.web.RequestMapping;
+import core.annotation.web.RequestMethod;
+import core.mvc.tobe.handler.HandlerExecution;
+import core.mvc.tobe.handler.HandlerKey;
+import core.mvc.tobe.handlermapping.HandlerMapping;
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.scanners.TypeAnnotationsScanner;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Arrays;
+import java.util.Map;
+
+public class AnnotationHandlerMapping implements HandlerMapping {
+    private Object[] basePackage;
+
+    private Map<HandlerKey, HandlerExecution> handlerExecutions = Maps.newHashMap();
+
+    public AnnotationHandlerMapping(Object... basePackage) {
+        this.basePackage = basePackage;
+    }
+
+    @Override
+    public void init() {
+        Arrays.stream(basePackage)
+                .forEach(basePackage -> executeComponentScan(basePackage));
+    }
+
+    @Override
+    public HandlerExecution findHandler(HttpServletRequest request) {
+        String requestUri = request.getRequestURI();
+
+        if(request.getMethod() == null){
+            return null;
+        }
+
+        RequestMethod method = RequestMethod.valueOf(request.getMethod().toUpperCase());
+        HandlerExecution handlerExecution = handlerExecutions.get(new HandlerKey(requestUri, method));
+        return handlerExecution;
+    }
+
+    @Override
+    public boolean hasHandler(HttpServletRequest request) {
+        return findHandler(request) != null;
+    }
+
+    private Object getNewInstance(Class clazz) {
+        Object controller = null;
+
+        try {
+            controller = clazz.newInstance();
+        } catch (InstantiationException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
```

Comment:
아래와 같이 축약이 가능하겠네요 :)

```suggestion
        } catch (InstantiationException | IllegalAccessException e) {
            e.printStackTrace();
        } 
```